var/spack/repos/builtin/packages/py-pyqt5/package.py

```diff
@@ -58,6 +62,13 @@ def configure_args(self):
             args.extend(['--qsci-api-destdir', self.prefix.share.qsci])
         return args
 
+    @run_after('install')
+    def extend_path_setup(self):
+        with working_dir(site_packages_dir):
+            with open('./PyQt5/__init__.py', 'a') as f:
+                f.write('from pkgutil import extend_path\n')
+                f.write('__path__ = extend_path(__path__, __name__)\n')
```

Comment:
Let's move this to the `SipPackage` base class. Same logic as I listed for `py-sip`, but use `self.spec['py-sip'].variants['module'].value`

Reply:
when we do so, we have to make sure it will run **before** make_qsci otherwise make_qsci will fail. The way that I achieved it was to put extend_path function before make_qsci. Let's see if it still works.

Reply:
just confirmed that `+qsci` variant builds fine with the current setup